### PR TITLE
updated token sale setup proxy and token collection issuer contract

### PR DESCRIPTION
**Pull request by ldgaetano**

### CODE REVIEW

File: src/main/scala/contracts/token_sale_setup_proxy/v1.0.0/ergoscript/Token_Sale_Setup_Proxy.es [hunk 1]
```diff
@@ -47,7 +48,7 @@
             }
 
             val validContract: Boolean = {
-                (tokenCollectionIssuerBoxOUT.propositionBytes == _TokenCollectionIssuerProxyContractBytes)
+                (tokenCollectionIssuerBoxOUT.propositionBytes == TokenCollectionIssuerProxyContractBytes)
```
Comment (Luivatra) [inline src/main/scala/contracts/token_sale_setup_proxy/v1.0.0/ergoscript/Token_Sale_Setup_Proxy.es]: probably just a copypaste error, think this should be TokenCollectionIssuerContractBytes

File: src/main/scala/contracts/token_sale_setup_proxy/v1.0.0/ergoscript/Token_Sale_Setup_Proxy.es [hunk 2]
```diff
@@ -69,11 +70,11 @@
...
-                    (tokenCollectionIssuerBoxOUT.R4[Int].get == collectionStandardVersion),
-                    (tokenCollectionIssuerBoxOUT.R5[Coll[Coll[Byte]]].get == collectionInfo),
-                    (tokenCollectionIssuerBoxOUT.R6[Coll[(Coll[Byte], Coll[Byte])]].get == collectionSocials),
-                    (tokenCollectionIssuerBoxOUT.R7[Long].get == collectionMintingExpiry),
-                    (tokenCollectionIssuerBoxOUT.R8[Coll[(Coll[Byte], Coll[Byte])]].get == collectionAdditionalInfo)
+                    (tokenCollectionIssuerBoxOUT.R4[Int].get == CollectionStandardVersion),
+                    (tokenCollectionIssuerBoxOUT.R5[Coll[Coll[Byte]]].get == CollectionInfo),
+                    (tokenCollectionIssuerBoxOUT.R6[Coll[(Coll[Byte], Coll[Byte])]].get == CollectionSocials),
+                    (tokenCollectionIssuerBoxOUT.R7[Long].get == CollectionMintingExpiry),
+                    (tokenCollectionIssuerBoxOUT.R8[Coll[(Coll[Byte], Coll[Byte])]].get == CollectionAdditionalInfo)
```
Comment (Luivatra) [inline src/main/scala/contracts/token_sale_setup_proxy/v1.0.0/ergoscript/Token_Sale_Setup_Proxy.es:77]: these values come from context vars, so they can be anything the tx operator wants

File: src/main/scala/contracts/token_collection_issuer_proxy/v1.0.0/ergoscript/Token_Collection_Issuer.es [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    // _MinerFee: Long
+
+    // ===== Context Extension Variables ===== //
+    val TokenColectionIssuanceContractBytes: Coll[Byte] = getVar[Coll[Byte]](0).get
+    val CollectionTokenVerboseName: Coll[Byte] = getVar[Coll[Byte]](1).get
+    val CollectionTokenDescription: Coll[Byte] = getVar[Coll[Byte]](2).get
+    val CollectionTokenNumberOfDecimals: Coll[Byte] = getVar[Coll[Byte]](3).get // Should be 0, encoded according to EIP-4
+    val CollectionTokenAssetType: Coll[Byte] = Coll(3.toByte, 2.toByte) // The asset type category should be Utility-Token (or NFT Artwork?)
+    val CollectionTokenPictureSHA256Hash: Coll[Byte] = Coll[Byte]() // Should just be empty Coll[Byte](), if Utility-Token
+    val CollectionTokenPictureLink: Coll[Byte] = Coll[Byte]() // Should just be empty Coll[Byte](), if Utility-Token
```
Comment (Luivatra) [inline src/main/scala/contracts/token_collection_issuer_proxy/v1.0.0/ergoscript/Token_Collection_Issuer.es:36]: The tx operator decides the context variables, so they can insert whatever they want

**Review by Luivatra — CHANGES_REQUESTED** [review]
I am a bit confused about your usage of context vars. what is stopping the tx operator from inserting garbage?

**Comment by ldgaetano** [comment]
I added the TxOperatorPK as a condition for singing and fixed the typos

**Pull request merged**